<a href="https://colab.research.google.com/github/fralfaro/MAT281/blob/main/docs/labs/lab_03.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


# MAT281 - Laboratorio N°03





**Objetivo**: Aplicar técnicas avanzadas de manipulación y análisis de datos con pandas sobre un conjunto real de datos de contenido de Netflix, reforzando buenas prácticas y métodos eficientes sin recurrir a `groupby`, `merge`, `pivot`, ni `join`.



**Dataset**:

Trabajaremos con el archivo `netflix_titles.csv`, que contiene información sobre los títulos disponibles en la plataforma Netflix hasta el año 2021.

| Variable       | Clase     | Descripción                                                                 |
|----------------|-----------|------------------------------------------------------------------------------|
| show_id        | caracter  | Identificador único del título en el catálogo de Netflix.                   |
| type           | caracter  | Tipo de contenido: 'Movie' o 'TV Show'.                                     |
| title          | caracter  | Título del contenido.                                                       |
| director       | caracter  | Nombre del director (puede ser nulo).                                       |
| cast           | caracter  | Lista de actores principales (puede ser nulo).                              |
| country        | caracter  | País o países donde se produjo el contenido.                                |
| date_added     | fecha     | Fecha en la que el título fue agregado al catálogo de Netflix.              |
| release_year   | entero    | Año de lanzamiento original del título.                                     |
| rating         | caracter  | Clasificación por edad (por ejemplo: 'PG-13', 'TV-MA').                      |
| duration       | caracter  | Duración del contenido (minutos o número de temporadas para series).        |
| listed_in      | caracter  | Categorías o géneros en los que está clasificado el contenido.              |
| description    | caracter  | Breve sinopsis del contenido.                                               |




In [47]:
import pandas as pd

# Cargar datos
df = pd.read_csv('https://raw.githubusercontent.com/fralfaro/MAT281/main/docs/labs/data/netflix_titles.csv')
df.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...


### Parte 1: Limpieza y preparación

1. Revisar y describir el dataset:

   * ¿Cuántas filas y columnas tiene?
   * ¿Qué tipos de datos hay?
   * ¿Cuántos valores nulos hay por columna?

2. Transformar la columna `date_added` a tipo fecha.

3. Crear columnas auxiliares con `assign`:

   * Año (`year_added`)
   * Mes (`month_added`)



Hay en total 8807 filas y 12 columnas. \
Hay datos tipo **object** y tipo **int64**. \
Las unicas columnas que tienen valores nulos son director con $2634$, cast con $825$, country	con $831$, date_added con $10$, rating con $4$ y duration con $3$, todas las demas columnas tienen $0$ valores nulos por columna.

In [48]:
#FIXME
#Parte 1
#1.
#Filas,Columnas
df.shape

(8807, 12)

In [49]:
#Tipos de datos
df.dtypes

show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration        object
listed_in       object
description     object
dtype: object

In [50]:
#Valores nulos por columna
df.isnull().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

In [51]:
#Punto 2
# cambiar formato de la fecha
import datetime
df['date'] = pd.to_datetime(df['date_added'], errors='coerce')
df.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description       date  
0  As her father nears the end of his life, filmm... 2021-09-25  
1  After crossing paths at a party, a Cape Town t... 2021-09-24  
2  To protect his family from a powerful drug lor... 2021-09-24  
3  Feuds, flirtations and toilet talk go down amo... 2021-09-24  
4  In a city of coaching centers known to train I... 2021-09-24

In [52]:
#Punto 3
df = df.assign(year_added = df['date'].dt.year,month_added = df['date'].dt.month)
df.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description       date  year_added  \
0  As her father nears the end of his life, filmm... 2021-09-25      2021.0   
1  After crossing paths at a party, a Cape Town t... 2021-09-24      2021.0   
2  To protect his family from a powerful drug lor... 2021-09-24      2021.0   
3  Feuds, flirtations and toilet talk go down amo... 2021-09-24      2021.0   
4  In a city of coaching centers known to train I... 2021-09-24      2021.0   

   month_added  
0          9.0  
1          9.0  
2          9.0  
3          9.0  
4          9.0

## Parte 2: Técnicas avanzadas de pandas

4. Utilizar `.loc` para seleccionar películas (`type == 'Movie'`) que fueron agregadas después del año 2018.

5. Utilizar `str.contains()` y `str.extract()`:

   * Filtrar títulos que contienen la palabra 'love' (sin distinguir mayúsculas/minúsculas).
   * Extraer la duración en minutos para las películas desde la columna `duration`.

6. Aplicar `explode()` sobre la columna `listed_in` para obtener una fila por cada género.

7. Obtener un top 10 de géneros más frecuentes utilizando `value_counts()`.

8. Aplicar `where()` y `mask()` para marcar las películas de más de 120 minutos como contenido largo en una nueva columna.

9. Utilizar `.loc` para filtrar películas que cumplen con:

   * Más de 100 minutos de duración.
   * Rating igual a `'R'`.
   * País igual a `'United States'`.

10. Utilizar `.style` para formatear visualmente el top 10 de películas más largas.

In [53]:
#FIXME
#Parte 2
#Punto 4
df_new = df.loc[(df['type']=='Movie') & (df['year_added']>2018)]
df_new.head()

show_id   type                             title  \
0       s1  Movie              Dick Johnson Is Dead   
6       s7  Movie  My Little Pony: A New Generation   
7       s8  Movie                           Sankofa   
9      s10  Movie                      The Starling   
12     s13  Movie                      Je Suis Karl   

                         director  \
0                 Kirsten Johnson   
6   Robert Cullen, José Luis Ucha   
7                    Haile Gerima   
9                  Theodore Melfi   
12            Christian Schwochow   

                                                 cast  \
0                                                 NaN   
6   Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
7   Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
9   Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   
12  Luna Wedler, Jannis Niewöhner, Milan Peschel, ...   

                                              country          date_added  \
0                                       United States  September 25, 2021   
6                                                 NaN  September 24, 2021   
7   United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
9                                       United States  September 24, 2021   
12                            Germany, Czech Republic  September 23, 2021   

    release_year rating duration  \
0           2020  PG-13   90 min   
6           2021     PG   91 min   
7           1993  TV-MA  125 min   
9           2021  PG-13  104 min   
12          2021  TV-MA  127 min   

                                           listed_in  \
0                                      Documentaries   
6                           Children & Family Movies   
7   Dramas, Independent Movies, International Movies   
9                                   Comedies, Dramas   
12                      Dramas, International Movies   

                                          description       date  year_added  \
0   As her father nears the end of his life, filmm... 2021-09-25      2021.0   
6   Equestria's divided. But a bright-eyed hero be... 2021-09-24      2021.0   
7   On a photo shoot in Ghana, an American model s... 2021-09-24      2021.0   
9   A woman adjusting to life after a loss contend... 2021-09-24      2021.0   
12  After most of her family is murdered in a terr... 2021-09-23      2021.0   

    month_added  
0           9.0  
6           9.0  
7           9.0  
9           9.0  
12          9.0

In [54]:
#Punto 5
df_new = df.loc[df['title'].str.lower().str.contains('love')]
df_new.head()

show_id     type                      title               director  \
25      s26  TV Show       Love on the Spectrum                    NaN   
158    s159    Movie    Love Don't Cost a Thing              Troy Byer   
159    s160    Movie             Love in a Puff         Pang Ho-cheung   
206    s207    Movie  LSD: Love, Sex Aur Dhokha       Dibakar Banerjee   
227    s228    Movie                Really Love  Angel Kristi Williams   

                                                  cast        country  \
25                                    Brooke Satchwell      Australia   
158  Nick Cannon, Christina Milian, Kenan Thompson,...  United States   
159  Miriam Chin Wah Yeung, Shawn Yue, Singh Hartih...      Hong Kong   
206  Nushrat Bharucha, Anshuman Jha, Neha Chauhan, ...          India   
227  Kofi Siriboe, Yootha Wong-Loi-Sing, Michael Ea...  United States   

             date_added  release_year rating   duration  \
25   September 21, 2021          2021  TV-14  2 Seasons   
158   September 1, 2021          2003  PG-13    101 min   
159   September 1, 2021          2010  TV-MA    103 min   
206     August 27, 2021          2010  TV-MA    112 min   
227     August 25, 2021          2020  TV-MA     95 min   

                                            listed_in  \
25     Docuseries, International TV Shows, Reality TV   
158                         Comedies, Romantic Movies   
159            Comedies, Dramas, International Movies   
206  Dramas, Independent Movies, International Movies   
227       Dramas, Independent Movies, Romantic Movies   

                                           description       date  year_added  \
25   Finding love can be hard for anyone. For young... 2021-09-21      2021.0   
158  A nerdy teen tries to make himself cool by ass... 2021-09-01      2021.0   
159  When the Hong Kong government enacts a ban on ... 2021-09-01      2021.0   
206  This provocative drama examines how the voyeur... 2021-08-27      2021.0   
227  A rising Black painter tries to break into a c... 2021-08-25      2021.0   

     month_added  
25           9.0  
158          9.0  
159          9.0  
206          8.0  
227          8.0

In [55]:
#Punto 5
movies_df = df.loc[df['type'] == 'Movie'].copy()
movies_df['duration_minutes'] = movies_df['duration'].str.extract('(\d+) min').astype(float)
display(movies_df.head())

<>:3: SyntaxWarning: invalid escape sequence '\d'
<>:3: SyntaxWarning: invalid escape sequence '\d'
/tmp/ipython-input-905417708.py:3: SyntaxWarning: invalid escape sequence '\d'
  movies_df['duration_minutes'] = movies_df['duration'].str.extract('(\d+) min').astype(float)


show_id   type                             title  \
0       s1  Movie              Dick Johnson Is Dead   
6       s7  Movie  My Little Pony: A New Generation   
7       s8  Movie                           Sankofa   
9      s10  Movie                      The Starling   
12     s13  Movie                      Je Suis Karl   

                         director  \
0                 Kirsten Johnson   
6   Robert Cullen, José Luis Ucha   
7                    Haile Gerima   
9                  Theodore Melfi   
12            Christian Schwochow   

                                                 cast  \
0                                                 NaN   
6   Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
7   Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
9   Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   
12  Luna Wedler, Jannis Niewöhner, Milan Peschel, ...   

                                              country          date_added  \
0                                       United States  September 25, 2021   
6                                                 NaN  September 24, 2021   
7   United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
9                                       United States  September 24, 2021   
12                            Germany, Czech Republic  September 23, 2021   

    release_year rating duration  \
0           2020  PG-13   90 min   
6           2021     PG   91 min   
7           1993  TV-MA  125 min   
9           2021  PG-13  104 min   
12          2021  TV-MA  127 min   

                                           listed_in  \
0                                      Documentaries   
6                           Children & Family Movies   
7   Dramas, Independent Movies, International Movies   
9                                   Comedies, Dramas   
12                      Dramas, International Movies   

                                          description       date  year_added  \
0   As her father nears the end of his life, filmm... 2021-09-25      2021.0   
6   Equestria's divided. But a bright-eyed hero be... 2021-09-24      2021.0   
7   On a photo shoot in Ghana, an American model s... 2021-09-24      2021.0   
9   A woman adjusting to life after a loss contend... 2021-09-24      2021.0   
12  After most of her family is murdered in a terr... 2021-09-23      2021.0   

    month_added  duration_minutes  
0           9.0              90.0  
6           9.0              91.0  
7           9.0             125.0  
9           9.0             104.0  
12          9.0             127.0

In [56]:
#Punto 6
genero_df = df.assign(listed_in=df['listed_in'].str.split(', ')).explode('listed_in')
genero_df.head()

show_id     type                 title         director  \
0      s1    Movie  Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show         Blood & Water              NaN   
1      s2  TV Show         Blood & Water              NaN   
1      s2  TV Show         Blood & Water              NaN   
2      s3  TV Show             Ganglands  Julien Leclercq   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   

           date_added  release_year rating   duration               listed_in  \
0  September 25, 2021          2020  PG-13     90 min           Documentaries   
1  September 24, 2021          2021  TV-MA  2 Seasons  International TV Shows   
1  September 24, 2021          2021  TV-MA  2 Seasons               TV Dramas   
1  September 24, 2021          2021  TV-MA  2 Seasons            TV Mysteries   
2  September 24, 2021          2021  TV-MA   1 Season          Crime TV Shows   

                                         description       date  year_added  \
0  As her father nears the end of his life, filmm... 2021-09-25      2021.0   
1  After crossing paths at a party, a Cape Town t... 2021-09-24      2021.0   
1  After crossing paths at a party, a Cape Town t... 2021-09-24      2021.0   
1  After crossing paths at a party, a Cape Town t... 2021-09-24      2021.0   
2  To protect his family from a powerful drug lor... 2021-09-24      2021.0   

   month_added  
0          9.0  
1          9.0  
1          9.0  
1          9.0  
2          9.0

In [57]:
#Punto 7
genero_df['listed_in'].value_counts().head(10)

listed_in
International Movies        2752
Dramas                      2427
Comedies                    1674
International TV Shows      1351
Documentaries                869
Action & Adventure           859
TV Dramas                    763
Independent Movies           756
Children & Family Movies     641
Romantic Movies              616
Name: count, dtype: int64

In [58]:
#Punto 8
movies_df['long_content'] = movies_df['duration_minutes'].where(movies_df['duration_minutes'] > 120, 'short')
movies_df['long_content'] = movies_df['long_content'].mask(movies_df['duration_minutes'] > 120, 'long')
movies_df.head()

show_id   type                             title  \
0       s1  Movie              Dick Johnson Is Dead   
6       s7  Movie  My Little Pony: A New Generation   
7       s8  Movie                           Sankofa   
9      s10  Movie                      The Starling   
12     s13  Movie                      Je Suis Karl   

                         director  \
0                 Kirsten Johnson   
6   Robert Cullen, José Luis Ucha   
7                    Haile Gerima   
9                  Theodore Melfi   
12            Christian Schwochow   

                                                 cast  \
0                                                 NaN   
6   Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
7   Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
9   Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   
12  Luna Wedler, Jannis Niewöhner, Milan Peschel, ...   

                                              country          date_added  \
0                                       United States  September 25, 2021   
6                                                 NaN  September 24, 2021   
7   United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
9                                       United States  September 24, 2021   
12                            Germany, Czech Republic  September 23, 2021   

    release_year rating duration  \
0           2020  PG-13   90 min   
6           2021     PG   91 min   
7           1993  TV-MA  125 min   
9           2021  PG-13  104 min   
12          2021  TV-MA  127 min   

                                           listed_in  \
0                                      Documentaries   
6                           Children & Family Movies   
7   Dramas, Independent Movies, International Movies   
9                                   Comedies, Dramas   
12                      Dramas, International Movies   

                                          description       date  year_added  \
0   As her father nears the end of his life, filmm... 2021-09-25      2021.0   
6   Equestria's divided. But a bright-eyed hero be... 2021-09-24      2021.0   
7   On a photo shoot in Ghana, an American model s... 2021-09-24      2021.0   
9   A woman adjusting to life after a loss contend... 2021-09-24      2021.0   
12  After most of her family is murdered in a terr... 2021-09-23      2021.0   

    month_added  duration_minutes long_content  
0           9.0              90.0        short  
6           9.0              91.0        short  
7           9.0             125.0         long  
9           9.0             104.0        short  
12          9.0             127.0         long

In [59]:
#Punto 9
#Primero agrego la columna duration_minutes al dataframe
df['duration_minutes'] = df['duration'].where(df['type'] == 'Movie').str.extract(r'(\d+)').astype(float)
#Luego saco la informacion
df_minute= df.loc[(df['type']=='Movie') & (df['duration_minutes']>100) & (df['rating']=='R') & (df['country']=='United States')]
df_minute.head()


show_id   type                        title               director  \
48      s49  Movie                 Training Day          Antoine Fuqua   
81      s82  Movie                         Kate  Cedric Nicolas-Troyan   
131    s132  Movie  Blade Runner: The Final Cut           Ridley Scott   
139    s140  Movie           Do the Right Thing              Spike Lee   
144    s145  Movie                  House Party        Reginald Hudlin   

                                                  cast        country  \
48   Denzel Washington, Ethan Hawke, Scott Glenn, T...  United States   
81   Mary Elizabeth Winstead, Jun Kunimura, Woody H...  United States   
131  Harrison Ford, Rutger Hauer, Sean Young, Edwar...  United States   
139  Danny Aiello, Ossie Davis, Ruby Dee, Richard E...  United States   
144  Christopher Reid, Christopher Martin, Robin Ha...  United States   

             date_added  release_year rating duration  \
48   September 16, 2021          2001      R  122 min   
81   September 10, 2021          2021      R  106 min   
131   September 1, 2021          1982      R  117 min   
139   September 1, 2021          1989      R  120 min   
144   September 1, 2021          1990      R  104 min   

                                           listed_in  \
48                                 Dramas, Thrillers   
81                                Action & Adventure   
131  Action & Adventure, Classic Movies, Cult Movies   
139                 Classic Movies, Comedies, Dramas   
144                            Comedies, Cult Movies   

                                           description       date  year_added  \
48   A rookie cop with one day to prove himself to ... 2021-09-16      2021.0   
81   Slipped a fatal poison on her final job, a rut... 2021-09-10      2021.0   
131  In a smog-choked dystopian Los Angeles, blade ... 2021-09-01      2021.0   
139  On a sweltering day in Brooklyn, simmering rac... 2021-09-01      2021.0   
144  Grounded by his strict father, Kid risks life ... 2021-09-01      2021.0   

     month_added  duration_minutes  
48           9.0             122.0  
81           9.0             106.0  
131          9.0             117.0  
139          9.0             120.0  
144          9.0             104.0

In [60]:
#Punto 10 Utilizar
top10_df=movies_df.sort_values('duration_minutes',ascending=False).head(10)
top10_df.style.bar(subset=['duration_minutes'], color='yellow', align='mid')



### Pregunta Desafío

11. ¿Cuáles son las combinaciones más frecuentes de género y rating en el dataset?
    (Sugerencia: utilizar `value_counts` con `subset=["genre", "rating"]` después de aplicar `explode()`).



### Bonus: Análisis de duplicados y limpieza

12. ¿Existen películas con el mismo nombre (`title`) pero con distinto año de lanzamiento (`release_year`)?
13. ¿Cuántos títulos únicos hay en total en la columna `title`?





In [61]:
#Punto 11
df.dropna(subset=['listed_in','rating'], inplace=True)
df['genre_list'] = df['listed_in'].str.split(',')
# Aplico explode para que cada genero tenga su propia fila
df_exploded=df.explode('genre_list')
df_exploded.rename(columns={'genre_list': 'genre'}, inplace=True)
top_combinacion=df_exploded.value_counts(subset=['genre', 'rating'])
top_combinacion.head()

genre                  rating
 International Movies  TV-MA     1074
                       TV-14     1022
Dramas                 TV-MA      616
                       TV-14      428
 TV Dramas             TV-MA      401
Name: count, dtype: int64

In [62]:
#Punto 12 Bonus
tduplicado_df=df[df.duplicated(subset=['title'], keep=False)]
tduplicado_df.head()

Empty DataFrame
Columns: [show_id, type, title, director, cast, country, date_added, release_year, rating, duration, listed_in, description, date, year_added, month_added, duration_minutes, genre_list]
Index: []

Se puede ver que no genera ninguna pelicula, por tanto no existen peliculas con el mismo nombre pero con distinto año de lanzamiento.

In [63]:
#Punto 13
unico_df=df['title'].nunique()
unico_df

8803

Hay en total $8803$ titulos unicos en la columna "title"